flytekit/core/python_function_task.py

```diff
@@ -349,15 +349,21 @@ def dynamic_execute(self, task_function: Callable, **kwargs) -> Any:
         raise ValueError(f"Invalid execution provided, execution state: {ctx.execution_state}")
 
     def _write_decks(self, native_inputs, native_outputs_as_map, ctx, new_user_params):
-        # These errors are raised if the source code can not be retrieved
-        with suppress(OSError, TypeError):
-            source_code = inspect.getsource(self._task_function)
-
+        if self._disable_deck is False:
             from flytekit.deck import Deck
-            from flytekit.deck.renderer import SourceCodeRenderer
+            from flytekit.deck.renderer import PythonDependencyRenderer
+
+            # These errors are raised if the source code can not be retrieved
+            with suppress(OSError, TypeError):
+                source_code = inspect.getsource(self._task_function)
+                from flytekit.deck.renderer import SourceCodeRenderer
+
+                source_code_deck = Deck("Source Code")
+                renderer = SourceCodeRenderer()
+                source_code_deck.append(renderer.to_html(source_code))
 
-            source_code_deck = Deck("Source Code")
-            renderer = SourceCodeRenderer()
-            source_code_deck.append(renderer.to_html(source_code))
+            python_dependencies_deck = Deck("Python Dependencies")
```

Comment:
Nit: Making this shorter looks better in the UI:

```suggestion
            python_dependencies_deck = Deck("Dependencies")
```


![Screenshot 2024-04-02 at 3 21 58 PM](https://github.com/flyteorg/flytekit/assets/5402633/590a0d1b-e2c2-4106-9c55-80d8216703c6)